influxql/ast.go

```diff
@@ -1133,6 +1133,23 @@ func (s *SelectStatement) validateAggregates(tr targetRequirement) error {
 				if err := s.validSelectWithAggregate(numAggregates); err != nil {
 					return err
 				}
+				// Validate that if they have a time dimension, they need a sub-call like min/max, etc.
+				if s.hasTimeDimensions(s.Condition) {
+					var valid bool
+					for _, a := range expr.Args {
+						if c, ok := a.(*Call); ok {
+							n := strings.ToLower(c.Name)
+							if n == "min" || n == "max" || n == "first" || n == "last" || n == "mean" || n == "sum" {
```

Comment:
Is this actually required?  Might be a bit brittle as we add new functions.  

I think `count` could be here as well.